src/libexpr/primops/fetchTree.cc

```diff
@@ -48,10 +51,40 @@ void emitTreeAttrs(
     v.attrs->sort();
 }
 
-static void prim_fetchTree(EvalState & state, const Pos & pos, Value * * args, Value & v)
+std::string fixURI(std::string uri, EvalState &state)
+{
+    state.checkURI(uri);
```

Comment:
We can remove `checkURI` since the check can easily be by-passed.

Reply:
Without that most of the tests fail like this:

```
    + nix eval --raw --expr 'builtins.fetchTarball https://github.com/NixOS/patchelf/archive/master.tar.gz' --impure --restrict-eval
    error: --- RestrictedPathError --- nix
    access to URI 'https://github.com/NixOS/patchelf/archive/master.tar.gz' is forbidden in restricted mode
```